org.scala-ide.sdt.core.tests/src/org/scalaide/core/testsetup/SDTTestUtils.scala

```diff
@@ -39,10 +39,11 @@ import org.eclipse.jdt.launching.JavaRuntime
 import org.scalaide.core.compiler.IScalaPresentationCompiler
 import org.scalaide.core.internal.project.ScalaProject
 
-/** Utility functions for setting up test projects.
+/**
+ * Utility functions for setting up test projects.
  *
  */
-object SDTTestUtils extends HasLogger {
+trait SDTTestUtils extends HasLogger {
```

Comment:
so the idea is following:
- let move a part which creates workspace to object
- the rest break up into functional parts:
  - ProjectUtils = project's creation, initialization and clean up
  - ErrorUtils = markers scanning on project, compile units
  - FileManipulationUtils = compile units, content replacement
  - maybe something else ?


2015-05-18 9:39 GMT+02:00 Simon Schäfer <notifications@github.com>:

> In
> org.scala-ide.sdt.core.tests/src/org/scalaide/core/testsetup/SDTTestUtils.scala
> <https://github.com/scala-ide/scala-ide/pull/948#discussion_r30486330>:
>
> >   *
> >   */
> > -object SDTTestUtils extends HasLogger {
> > +trait SDTTestUtils extends HasLogger {
>
> If there is something you don't like with the test suite, please change it
> and educate us. I just don't want to introduce new types everywhere without
> seeing a compelling reason.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scala-ide/scala-ide/pull/948/files#r30486330>.
>